carte_cli/extractor/glue_extractor.py

```diff
@@ -43,22 +53,30 @@ def _get_extract_iter(self) -> Iterator[TableMetadata]:
             ):
                 continue
 
-            for column in row["StorageDescriptor"]["Columns"] + row.get(
-                "PartitionKeys", []
-            ):
-                columns.append(
-                    ColumnMetadata(
-                        name=column["Name"],
-                        column_type=column["Type"],
-                        description=None,
+            is_view = row.get("TableType") == "VIRTUAL_VIEW"
+
+            if is_view:
+                for column in row["StorageDescriptor"]["Columns"]:
+                    columns.append(
+                        ColumnMetadata(
+                            name=column["Name"],
+                            column_type=column["Type"],
+                            description=None,
+                        )
+                    )
+            else:
+                # This solution is more robust for Delta and Parquet tables.
+                # Both table types has these fields
+                for column in row["Parameters"]["spark.sql.sources.schema"]["fields"]:
```

Comment:
Could we use this also for normal tables or does only `row["StorageDescriptor"]["Columns"]` work?

Reply:
So this will work now for both `Parquet` and `Delta` tables. Only the views don't work with this. The condition above is for the views only.